api/v1alpha1/packagebundle.go

```diff
@@ -23,20 +25,22 @@ func (config *PackageBundle) ExpectedKind() string {
 	return PackageBundleKind
 }
 
-func (config *PackageBundle) FindSource(pkgName, pkgVersion string) (retSource PackageOCISource, version string, err error) {
+func (config *PackageBundle) FindSource(pkgName, pkgVersion string) (retSource PackageOCISource, err error) {
 	for _, pkg := range config.Spec.Packages {
 		if pkg.Name == pkgName {
 			source := pkg.Source
 			for _, version := range source.Versions {
-				if version.Name == pkgVersion || version.Digest == pkgVersion {
-					retSource = PackageOCISource{Registry: source.Registry, Repository: source.Repository, Digest: version.Digest}
-					return retSource, version.Name, nil
+				//We do not sort before getting `latest` because there will be only a single version per release in normal cases. For edge cases which may require multiple
+				//versions, the order in the file will be ordered according to what we want `latest` to point to
+				if version.Name == pkgVersion || version.Digest == pkgVersion || pkgVersion == Latest {
```

Comment:
Not asking you to change this, but should we be doing a case-insensitive comparison? At least for the digest, case definitely doesn't matter, and I think we can make a case that case shouldn't matter in either of the other cases as well.

Are there any cases where case would matter for this purpose? I can't think of any. Also, at least in the case of a digest, I can imagine someone specifying the digest in whatever case we're not expecting.